src/adapters/openai/helpers/format-output.ts

```diff
@@ -0,0 +1,138 @@
+  private static async _formatWithO1Mini(text: string, client: OpenAI): Promise<string> {
+    try {
+      const response = await client.chat.completions.create({
+        model: "gpt-4o",
+        messages: [
+          {
+            role: "system",
+            content: `You are an authoritative technical writer. Format and rephrase the input text following these rules:
+
+1. Writing Style:
+- Use direct, authoritative statements
+- Remove phrases like "Based on the context", "I think", "It appears"
+- Start sentences with action verbs when possible
+- Be concise.
+- Do Not Add titles to your responses.
+- Maintain technical accuracy
+- Keep existing citations intact
+- Expand on technical concepts when relevant
+- Include detailed explanations for complex topics
+- Provide context for technical decisions
+- Add clarifying examples where helpful
+
+2. Formatting:
+For code:
+\`\`\`language
+code here
+\`\`\`
+
+For lists:
+- Main point with detailed explanation
+  - Sub-detail with context
+  - Sub-detail with examples
+
+For headings:
+# Main Topic with Context
+
+## Sub-topic with Details
+
+For paragraphs:
+One clear statement per paragraph with supporting details.
+
+Add empty line between paragraphs.
+
+Return only the formatted text with improved style.`,
+          },
+          {
+            role: "user",
+            content: `Format and improve this text:\n\n${text}`,
+          },
+        ],
+        temperature: 0.1,
+        max_tokens: 2000,
+      });
+
+      const formattedText = response.choices[0]?.message?.content;
+      if (!formattedText) {
+        return this._fallbackFormatting(text);
+      }
+
+      return formattedText;
+    } catch (error) {
+      console.error("Error formatting with o1-mini:", error);
+      return this._fallbackFormatting(text);
+    }
+  }
+
+  private static _fallbackFormatting(text: string): string {
+    let formattedText = text;
+
+    // Format code blocks
+    formattedText = formattedText.replace(/```([\s\S]*?)```/g, (match, code) => {
+      const trimmedCode = code.trim();
+      return `\n\n\`\`\`\n${trimmedCode}\n\`\`\`\n\n`;
+    });
+
+    // Format inline code
+    formattedText = formattedText.replace(/`([^`]+)`/g, (match) => `\n${match}\n`);
+
+    // Format lists
+    formattedText = formattedText.replace(/^[-*]\s/gm, "\n- ");
+    formattedText = formattedText.replace(/^\d+\.\s/gm, (match) => `\n${match}`);
+
+    // Format headings
+    formattedText = formattedText.replace(/^(#{1,6}\s.*?)$/gm, "\n$1\n");
+
+    // Format paragraphs with improved spacing
+    formattedText = formattedText
+      .split("\n")
+      .map((line) => line.trim())
+      .filter((line) => line.length > 0)
+      .join("\n\n");
+
+    // Clean up multiple newlines while preserving paragraph spacing
+    formattedText = formattedText.replace(/\n{3,}/g, "\n\n");
+
+    return formattedText.trim();
+  }
+
+  private static _removeWeakPhrases(text: string): string {
```

Comment:
I think this is only viable with temperature 0 etc. couldn't this also cause a problem with quotes? If the text is quoted from elsewhere and it contains one of these expressions it might cause problems.

Reply:
doesn't this remove uncertainty and caution from the statements that would follow these phrases? Isn't an LLM likely to interpret these strings as indicating subjectivity, I'd think that these would give key context that we are not 100% on the following fact.

I see it used at the end of `createCompletion` which would imply it's stripping these from final output that we receive as well as completions sent between llm calls, is that right?